Bluetooth: Controller: Support for Periodic Sync Receive enable

cvinayak opened the pull request:
Initial support for Periodic Advertising Synchronization
Receive Enable command.

Update support for Periodic Advertising Synchronization
Receive Enable command, so that PDUs are received so that
contents of Extended Common Payload Format is parsed and
process for information like Channel Map Update and BIGInfo.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Add defines for Periodic Advertising Receive Enable

Add defines for Periodic Advertising Receive Enable.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Initial support for Periodic Sync Receive enable

Initial support for Periodic Advertising Synchronization
Receive Enable command.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Update support for Periodic Sync Receive enable

Update support for Periodic Advertising Synchronization
Receive Enable command, so that PDUs are received so that
contents of Extended Common Payload Format is parsed and
process for information like Channel Map Update and BIGInfo.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
tests: Bluetooth: bsim: Test updated to use Periodic Sync Recv Enable

Updated the ISO BabbleSim test to use Periodic Sync Receive
enable command to disable Periodic Sync reports and yet
receive the BIGInfo report.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -3551,6 +3551,22 @@ static void le_per_adv_recv_enable(struct net_buf *buf, struct net_buf **evt)
+	if (!status) {
+		if (cmd->enable &
+		    BT_HCI_LE_SET_PER_ADV_RECV_ENABLE_FILTER_DUPLICATE) {
+			if (!dup_scan || (dup_count < 0)) {
```
```suggestion
			if (!dup_scan || (dup_count == DUP_FILTER_DISABLED)) {
```

cvinayak replied:
Fixed

Thalley reviewed (changes requested):
Looks like there is a significant overlap between this PR and https://github.com/zephyrproject-rtos/zephyr/pull/40872. 

Not sure which commits belongs to which PR, and as such I will not review this PR any further until this has been cleaned up.

cvinayak committed:
Bluetooth: Controller: Use DUP_FILTER_DISABLED define

Use DUP_FILTER_DISABLED define instead of magic value to
represent duplicate filtering being disabled.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
> Not sure which commits belongs to which PR, and as such I will not review this PR any further until this has been cleaned up.

This PR blocks #40872.

I will address the comments posted in #40872 here and have this PR merged first.

cvinayak committed:
Bluetooth: Controller: Rename drop to accept to avoid negations

Rename the variable drop to accept to avoid multiple use of
negations in the implementation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley commented:
> > Not sure which commits belongs to which PR, and as such I will not review this PR any further until this has been cleaned up.
> 
> This PR blocks #40872.
> 
> I will address the comments posted in #40872 here and have this PR merged first.

That also ended up being my conclusion, and that I had simply reviewed the wrong PR first :)

cvinayak committed:
Bluetooth: Controller: Minor changes related to review comments

Minor changes related to review comments.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
@Thalley I have updated this PR with the changes relavent here, and rebased #40872 so that you can know which ones are resolved. Once this PR is merged, I will continue on rest of the comment in that PR.

ppryga-nordic reviewed (changes requested):
Single minor thing that could be improved.

BTW. It is a bit uncomfortable to review a PR where comments are addressed later in separate commit.

cvinayak commented on include/bluetooth/hci.h:
```diff
@@ -1650,6 +1650,9 @@ struct bt_hci_rp_le_read_ant_info {
 	uint8_t max_cte_len;
 };
 
+#define BT_HCI_LE_SET_PER_ADV_RECV_ENABLE_ENABLE           BIT(0)
```
This is something done in nRFx MDK often for register values, example `CCM_ENABLE_ENABLE_Enabled` !  hence I took the liberty to do similar here.

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_sync.c:
```diff
@@ -349,8 +349,27 @@ uint8_t ll_sync_terminate(uint16_t handle)
+
+	/* Reporting enabled/disabled */
+	lll = &sync->lll;
+	lll->is_rx_enabled = (enable &
```
`enable` is 1 octet HCI command parameter, and would like to keep it as-is close to specification defined 1 octet value.

ppryga replied:
Ok let is be as is, but could you provide a doxygen-like function description that includes information about parameters. Then users would be able to find out the purpose of the parameter without going through implementation.

cvinayak replied:
done

cvinayak committed:
Bluetooth: Controller: Apply suggestions from code review

Apply suggestions from code review.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

ppryga-nordic reviewed (approved):
LGTM
